### PR TITLE
Revert 174 to enable updates and bump react to 16

### PR DESCRIPTION
**Pull request by gwwar**
We were maintaining a 2.x line for React 16, now that this is in Calypso, let's bump the React version on notifications-panel master.

This also reverts esnext changes to notifications-panel since https://github.com/Automattic/wp-calypso/pull/19698 has not landed yet and is blocking an update of notifications-panel in Calypso.

@ockham let us know when you're ready for another try.

### Testing Instructions
- `npm install`
- `npm start`
- Local testing environment loads as expected.

### CODE REVIEW

**Comment by gwwar** [comment]
Local dev looks a bit messed up, did we test the react branch locally before?

**Comment by gwwar** [comment]
Let me try again from the feature branch

**Comment by ockham** [comment]
> @ockham let us know when you're ready for another try.

Will do! Sorry for the extra work I caused :slightly_frowning_face:
